src/neural/onnx/converter.cc

```diff
@@ -186,12 +236,121 @@ void Converter::AddStdInitializers(OnnxBuilder* builder) {
                           Int64OnnxConst({-1, NumFilters() * 2, 1, 1}, {4}));
 }
 
+std::string Converter::MakeEncoderLayer(
+    OnnxBuilder* builder, const LegacyWeights::EncoderLayer& layer,
+    int embedding_size, int heads, const std::string& encoder_in,
+    const std::string& name) {
+  const int d_model = layer.mha.q_b.size();
+  const int depth = d_model / heads;
+
+  auto mha_shape =
+      builder->AddInitializer("/const" + name + "/mha/shape",
+                              Int64OnnxConst({-1, 64, heads, depth}, {4}));
+
+  auto flow = builder->MatMul(
+      name + "/mha/Q/w", encoder_in,
+      *GetWeghtsConverter(layer.mha.q_w, {embedding_size, d_model}, {1, 0}));
+  flow = builder->Add(name + "/mha/Q/b", flow,
+                      *GetWeghtsConverter(layer.mha.q_b, {d_model}));
```

Comment:
Hmmm I'm not sure that `operator*` of temporary object will extent it's lifetime...

This surely doesn't work:
```
const X& f(const X& x) { return x; }
const auto& t = f( X() );    // X() will be destroyed immideately.
```

This surely works:
```
const auto& t = std::make_unique<X>();   // std::unique_ptr lifetime is extended
```

About this I'm not sure, will research tomorrow:
```
const auto& t = *std::make_unique<X>(); 
```

Reply:
I'm not sure I understand the concern - we only need the temporary to be valid until the end of the function call, no?

Reply:
Yes we do, I wasn't sure that the nested temporary object would do it, but apparently it would, so we are good here.